includes/gateway/class-omise-payment-promptpay.php

```diff
@@ -0,0 +1,207 @@
+	}
+
+	/**
+	 * @see WC_Settings_API::init_form_fields()
+	 * @see woocommerce/includes/abstracts/abstract-wc-settings-api.php
+	 */
+	public function init_form_fields() {
+		$this->form_fields = array(
+			'enabled' => array(
+				'title'   => __( 'Enable/Disable', 'omise' ),
+				'type'    => 'checkbox',
+				'label'   => __( 'Enable Omise PromptPay Payment', 'omise' ),
+				'default' => 'no'
+			),
+
+			'title' => array(
+				'title'       => __( 'Title', 'omise' ),
+				'type'        => 'text',
+				'description' => __( 'This controls the title the user sees during checkout.', 'omise' ),
+				'default'     => __( 'PromptPay', 'omise' ),
+			),
+
+			'description' => array(
+				'title'       => __( 'Description', 'omise' ),
+				'type'        => 'textarea',
+				'description' => __( 'This controls the description the user sees during checkout.', 'omise' )
+			),
+		);
+	}
+
+	/**
+	 * @param WC_Order $order
+	 *
+	 * @see   woocommerce/templates/emails/email-order-details.php
+	 * @see   woocommerce/templates/emails/plain/email-order-details.php
+	 */
+	public function email_qrcode( $order ) {
+		if ( $this->id == $order->get_payment_method() ) {
+			$this->display_qrcode( $order, 'email' );
+		}
+	}
+
+	/**
+	 * @param int|WC_Order $order
+	 * @param string       $context  pass 'email' value through this argument only for 'sending out an email' case.
+	 */
+	public function display_qrcode( $order, $context = 'view' ) {
+		if ( ! $this->load_order( $order ) ) {
+			return;
+		}
+
+		$charge = OmiseCharge::retrieve( $this->get_charge_id_from_order() );
+		$qrcode = $charge['source']['scannable_code']['image']['download_uri'];
+
+		$expires_datetime = new WC_DateTime( $charge['expires_at'], new DateTimeZone( 'UTC' ) );
+		$expires_datetime->set_utc_offset( wc_timezone_offset() );
+
+		if ( 'view' === $context ) : ?>
+			<div id="omise-offline-additional-details" class="omise omise-additional-payment-details-box omise-promptpay-details" <?php echo 'email' === $context ? 'style="margin-bottom: 4em; text-align:center;"' : ''; ?>>
+				<p><?php echo __( 'Scan the QR code to pay', 'omise' ); ?></p>
+				<div class="omise omise-promptpay-qrcode">
+					<img style="margin: 0 auto;" src="<?php echo $qrcode; ?>" alt="Omise QR code ID: <?php echo $charge['source']['scannable_code']['image']['id']; ?>">
+				</div>
+				<div>
+					<?php echo __( 'Payment expires in: ', 'omise' ); ?>
+					<?php echo wc_format_datetime( $expires_datetime, wc_date_format() ); ?>
+					<?php echo wc_format_datetime( $expires_datetime, wc_time_format() ); ?>
+				</div>
+
+				<div id="omise-offline-payment-timeout" style="margin-top: 2em; display: none;">
+					<p><button id="omise-offline-payment-refresh-status">refresh status</button></p>
+				</div>
+			</div>
+
+			<div id="omise-offline-payment-result" class="omise-additional-payment-details-box" style="display: none;"></div>
+
+			<script>
```

Comment:
Note: this Javascript might need to be relocated to another file in the future once the same solution gets applied to PayNow payment method (or it may not possible, still testing the idea – however, it will be out of scope from PromptPay so I skip it)